### ISSUE TITLE
Fix `property-no-unknown` false positives for scroll-driven animations

### ISSUE THREAD

**Issue by renato-bohler** [stylelint/stylelint#7088]
### What minimal example or steps are needed to reproduce the bug?

```css
.view-timeline-works {
	view-timeline: --my-name block;
	view-timeline-axis: block;
	view-timeline-inset: 200px;
	view-timeline-name: --my-name;
}

.animation-range-works {
	animation-range: contain 100px;
	animation-range-start: contain;
	animation-range-end: 100px;
}

.animation-timeline-does-not-work {
	animation-timeline: view();
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-standard"
  ]
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0A3AljA7gWgC5oFsYAbNAOxm0wgCcBrAZwAJgAdM13dLPQk8mPCbZsBAJ7YyAQyJMARsQhg6AbnaduOfEVIVsUgB5oGQhUtXquGLX12VyDGLiEAmAAxuADgbUcrPbX49aSIhEXFJGRhfAF91MiQpMkIpfAgybBokgHNKanpmNj8klLSMrLJcoUgyXClyJgBGD29fThKCVLR0zJzKBjqaZyYauvI2uuTOst7KyhgyKCFmrx92OI4Ejq6ewLtsKAgYBkkIXCpaOhZLbZm9gSFNAAoASlj2EAAaEAAzNGIYAA5KIIEAwAwyTwApBgBgML7gdJ-bKgopMJisMEGXALKAMTFCADa7HR6MxAzEAN05xqyOwAySUCkNCgmJJTAAuusEbS0NkAGK0aaggBWDHSCNgnnhiDRGJAFKp5FwBPlxFSxxVX3Z5NwlKCNKRfPpkyZLNVmPVOIGbLIMRAMSAA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "15.10.2",
  "stylelint-config-standard": "34.0.0"
}
```

### What did you expect to happen?

The `animation-timeline` property to not be unknown.

### What actually happened?

The following error gets thrown:

```
Unexpected unknown property "animation-timeline"  property-no-unknown
```

### Do you have a proposal to fix the bug?

Add `animation-timeline` to `lib/reference/properties.js`? Though I'm not familiar with the source for this library, so I don't understand why `view-timeline` and `animation-range` properties are not unknown if they don't seem to be defined there.

I also don't comprehend why the `view` function is not being flagged as unknown as I don't see it being declared in `lib/reference/functions.js`.

So maybe we should add those as well? If that's the case, I'm happy to open a PR.

**Comment by renato-bohler** [comment]
For more context, in case anyone is curious, I'm using this on my portfolio: https://github.com/renato-bohler/renato-bohler.github.io/pull/92 and [this](https://github.com/renato-bohler/renato-bohler.github.io/actions/runs/5603064870/jobs/10249160987) is the GH Actions run that's failing.

**Comment by ybiquitous** [comment]
@renato-bohler Thanks for opening this issue with a reproducible demo!

The `property-no-unknown` rule depends on the [`known-css-properties`](https://github.com/known-css/known-css-properties) package, so changes to upstream would be best.

https://github.com/stylelint/stylelint/blob/f383a26b48b85bb344a61282af7837d7b3795880/lib/rules/property-no-unknown/index.js#L7

Could you please consider feedback to the `known-css-properties` repository if you're interested?

**Comment by ybiquitous** [comment]
FYI. The spec is here: https://drafts.csswg.org/scroll-animations/

**Comment by renato-bohler** [comment]
Hmm interesting, I'll definitely forward this to `known-css-properties`. Should we keep this issue open in the meantime? I can open a PR updating the version of this dependency whenever that gets done.

**Comment by ybiquitous** [comment]
> Should we keep this issue open in the meantime?

Yes! Let's keep it open so other people can find the same problem.

**Comment by jeddy3** [comment]
> The spec is here: https://drafts.csswg.org/scroll-animations/

That's a fascinating CSS feature. The actual [`animation-timeline` property is defined](https://drafts.csswg.org/css-animations-2/#animation-timeline) in [this spec](https://drafts.csswg.org/css-animations-2). There's also a [decent write-up on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-timeline). It's the first instance I've seen of a [`<dashed-ident>`](https://drafts.csswg.org/css-values-4/#typedef-dashed-ident) for a named `<custom-ident>`.

> I also don't comprehend why the `view` function is not being flagged as unknown as I don't see it being declared in `lib/reference/functions.js`.

As with the [known-css-properties](https://github.com/known-css/known-css-properties) package, we use [an upstream package](https://www.npmjs.com/package/css-functions-list) for functions (and it [includes `view`](https://github.com/niksy/css-functions-list/blob/master/index.json#L125)).

> The property-no-unknown rule depends on the [known-css-properties](https://github.com/known-css/known-css-properties) package, so changes to upstream would be best

Agreed. @renato-bohler In the meantime, you can workaround the issue using the [`ignoreProperties`](https://stylelint.io/user-guide/rules/property-no-unknown/#ignoreproperties-regex-regex-string) secondary option:

```json
{
  "extends": [
    "stylelint-config-standard"
  ],
  "rules": {
    "property-no-unknown": [true, { "ignoreProperties": ["animation-timeline"] }]
  }
}
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0A3AljA7gWgC5oFsYAbNAOxm0wgCcBrAZwAJgAdM13dLPQk8mPCbZsBAJ7YyAQyJMARsQhg6AbnaduOfEVIVsUgB5oGQhUtXquGLX12VyDGLiEAmAAxuADgbUcrPbX49aSIhEXFJGRhfAF91MiQpMkIpfAgybBokgHNKanpmNj8klLSMrLJcoUgyXClyJgBGD29fThKCVLR0zJzKBjqaZyYauvI2uuTOst7KyhgyKCFmrx92OI4Ejq6ewLtsKAgYBkkIXCpaOhZLbZm9gSFNAAoASlj2EAAaEAAzNGIYAA5KIIEAwAwyTwApBgBgML7gdJ-bKgopMJisMEGXALKAMTFCADa7HR6MxAzEAN05xqyOwAySUCkNCgmJJTAAup92ZiaABXAH4hDXMikjEgTw0CCeGBDCRkCDYPlkOgKzAcYWE3D8mCfFjitDZBU0GAABSlMqGGCFRMxt26GXuFExHKYMQ57I2MQRtMNADFaNNQQArBjpBGwTzwxBo8UUqnkXAE8XEVLHJNfHkgeNBGlIw30yZMlnJzGpnEDNlkb0xIA).
